test/python/primitives/test_backend_estimator.py

```diff
@@ -504,7 +504,7 @@ def test_dynamic_circuit(self):
             estimator = BackendEstimator(backend, skip_transpilation=True)
             estimator.set_transpile_options(seed_transpiler=15)
             result = estimator.run(qc, observable).result()
-        self.assertAlmostEqual(result.values[0], 0, places=1)
+        self.assertAlmostEqual(float(result.values[0]), -1.0, places=1)
```

Comment:
Should this have changed?  I feel like it still ought to be averaging to 0.  Maybe a bug in Aer?

Reply:
It shouldn't have changed as the condition should be identical. It would either be an aer bug or an issue in the primitive implementation (since the `BackendEstimatorV2` version didn't require any update). I opted to just roll will it here because this test will also be removed with the BackendEstimator v1 estimator implementation and the v2 tests didn't need any changes.

Reply:
Sounds like it must be a bug - I feel like we should either delete this test (and the sampler one) slightly early, or put a big comment on it saying that we're asserting something incorrect.  I don't like merging a PR that encodes an incorrect test, even if we're expecting it to be ephemeral.